src/mmrelay/matrix/relay.py

```
@@ -0,0 +1,436 @@
+                await asyncio.sleep(total_delay)
+            else:
+                facade.logger.exception(
+                    f"Error sending message to Matrix room {room_id} after {max_retries + 1} attempts"
+                )
+        else:
+            return response
+
+    return None
+
+
+async def matrix_relay(
```

Ruff (0.15.9)

[warning] Too many return statements (7 > 6)

(PLR0911)